firebase-firestore/src/main/java/com/google/firebase/firestore/util/AsyncQueue.java

```diff
@@ -471,19 +479,28 @@ public void enqueueAndForget(Runnable task) {
    * @return A DelayedTask instance that can be used for cancellation.
    */
   public DelayedTask enqueueAfterDelay(TimerId timerId, long delayMs, Runnable task) {
-    // While not necessarily harmful, we currently don't expect to have multiple tasks with the same
-    // timer id in the queue, so defensively reject them.
-    hardAssert(
-        !containsDelayedTask(timerId),
-        "Attempted to schedule multiple operations with timer id %s.",
-        timerId);
-
     DelayedTask delayedTask = createAndScheduleDelayedTask(timerId, delayMs, task);
     delayedTasks.add(delayedTask);
 
+    // Fast-forward delays for timerIds that have been overridden.
+    for (TimerId timerIdToSkip : timerIdsToSkip) {
+      if (delayedTask.timerId == timerIdToSkip) {
+        delayedTask.skipDelay();
+      }
+    }
     return delayedTask;
   }
 
+  /**
+   * For Tests: Skip all delays for a timer id.
```

Comment:
Maybe say "Skip all subsequent delays ..." to make it clear that it won't fast-forward already queued delayed operations.